Add Personal Campaign Page Admin UI Screen

kcristiano opened the pull request:
Overview
----------------------------------------
Recreate core screens as Search Displays
https://lab.civicrm.org/dev/core/-/issues/3912


Before
----------------------------------------
PCP Pages admin was the old UI

After
----------------------------------------
PCP Pages admin is the new UI

Technical Details
----------------------------------------
Improvements:  Edit, Approve and Reject links are created with custom URLs.  We should add to pcp.xml

Comments
----------------------------------------
Ready for review

kcristiano committed:
add Personal Campaign Page Admin UI Screen

kcristiano committed:
update path to replce admin screen

samuelsov commented on ext/civicrm_admin_ui/managed/SavedSearch_Personal_Campaign_Pages.mgd.php:
```diff
@@ -0,0 +1,267 @@
+          'classes' => [
+            'table',
+            'table-striped',
+          ],
```
To be consistent with others pages, it is good to add a css rule to greyed disabled rows:
```
'cssRules' => [
            [
              'disabled',
              'is_active',
              '=',
              FALSE,
            ],
          ],
```

samuelsov commented on ext/civicrm_admin_ui/ang/afsearchPersonalCampaignPages.aff.html:
```diff
@@ -0,0 +1,8 @@
+<div af-fieldset="" af-title="Personal Campaign Pages">
+  <div class="af-container af-layout-cols" af-title="Find Campaign Pages">
```
af-layout-inline and no title is more consistent with other screens.

samuelsov commented on ext/civicrm_admin_ui/managed/SavedSearch_Personal_Campaign_Pages.mgd.php:
```diff
@@ -0,0 +1,267 @@
+                  'style' => 'default',
+                  'path' => '',
+                  'action' => '',
+                  'condition' => [],
```
Show enable/disable action only if it makes sense:
```
'condition' => ['is_active', '=', TRUE],
```

samuelsov commented on ext/civicrm_admin_ui/managed/SavedSearch_Personal_Campaign_Pages.mgd.php:
```diff
@@ -0,0 +1,267 @@
+                  'icon' => 'fa-toggle-on',
+                  'text' => E::ts('Enable'),
+                  'style' => 'default',
+                  'condition' => [],
```
Show enable/disable action only if it makes sense:
```
'condition' => ['is_active', '=', FALSE],
```

samuelsov reviewed (commented):
@kcristiano Here a couple of thing that could be made more consistent

kcristiano committed:
show enable and disable buttons conditionally, formatting fixes

kcristiano commented:
Thanks @samuelsov

samuelsov commented:
Good for me.

colemanw commented:
Seems to be a conflict in the paths for approve/reject.

totten commented:
civibot, test this please

kcristiano committed:
change edit menu item to new PCP Entity update/edit

kcristiano commented:
@colemanw I updated PCP.xml for the update/edit action and regenerated the DAO .  I've switched over to use the new link.

This should now be ready.

colemanw commented on ext/civicrm_admin_ui/managed/SavedSearch_Personal_Campaign_Pages.mgd.php:
```diff
@@ -0,0 +1,284 @@
+                  'condition' => [],
+                ],
+                [
+                  'path' => 'civicrm/admin/pcp?action=revert&id=[id]',
```
This won't work because it has the same path as the page.

kcristiano replied:
@colemanw Thanks.  That makes sense.  Do we need to create a new page to handle reject/approve?

colemanw replied:
I think we briefly discussed yesterday the idea of making it a bulk action. That would involve adding a couple entity-specific api actions and then implementing `hook_civicrm_searchKitTasks`. Is that something you'd like to take a shot at or should I?

kcristiano replied:
@colemanw I'd like to give it a try - Just not sure when I can get it done

colemanw replied:
Ok if you need any help let me know @kcristiano

colemanw commented:
@kcristiano I'm afraid these schema changes will need to be done in the new way. See https://civicrm.org/blog/colemanw/what-xml-where-did-dao-go

colemanw commented:
@kcristiano I've rebased this now that  https://github.com/civicrm/civicrm-core/pull/30894 is merged.

aydun commented:
Test this please

mlutfy commented:
I tested on Standalone and this has 2 OKs, so I will merge.

It looks like this:

![image](https://github.com/user-attachments/assets/a7c9dd79-97d1-4386-8c11-5b74130079b2)

The second title presumably from `af-title` seems redundant and not translated, but maybe that's the case for most AdminUI forms (and maybe the title is required for CMS integrations?).

In any case, I think it's better merged, than not merged. We can continue tweaking later.

colemanw commented:
Thanks @mlutfy
